policy/reviewer/reviewer.go

```diff
@@ -177,42 +195,54 @@ func SelectReviewers(ctx context.Context, prctx pull.Context, result common.Resu
 
 		if child.ReviewRequestRule.Admins {
 			logger.Debug().Msg("Selecting from admins for review")
-			admins, err := selectAdmins(prctx)
+			adminTeams, err := selectAdmins(prctx)
 			if err != nil {
-				return nil, errors.Wrap(err, "failed to select admins")
+				return nil, nil, errors.Wrap(err, "failed to select admins")
 			}
-
-			for _, admin := range admins {
-				allUsers[admin] = struct{}{}
+			for team, members := range adminTeams {
+				allTeamsWithUsers[team] = members
+				for _, admin := range members {
+					allUsers[admin] = struct{}{}
+				}
 			}
 		}
 
-		var possibleReviewers []string
-		for u := range allUsers {
-			// Remove the author and any users who aren't collaborators
-			// since github will fail to assign _anyone_ if the request contains one of these
-			_, ok := collaboratorsToConsider[u]
-			if u != prctx.Author() && ok {
-				possibleReviewers = append(possibleReviewers, u)
+		allUsersKeys := make([]string, 0, len(allUsers))
+		for k := range allUsers {
+			allUsersKeys = append(allUsersKeys, k)
+		}
+
+		possibleReviewers := getPossibleReviewers(prctx, allUsersKeys, collaboratorsToConsider)
+		if child.GetMode() == common.RequestModePreferTeams {
+			var possibleTeams []string
+			teamMembersCount := 0
+			for team, users := range allTeamsWithUsers {
+				possibleReviewersInTeamCount := len(getPossibleReviewers(prctx, users, collaboratorsToConsider))
+				if possibleReviewersInTeamCount > 0 {
+					possibleTeams = append(possibleTeams, team)
+					teamMembersCount += possibleReviewersInTeamCount
+				}
+			}
+			if len(possibleTeams) > 0 && teamMembersCount >= child.ReviewRequestRule.RequiredCount {
```

Comment:
I don't think it's necessary to compare the total count of team members to the required count. The existing code for user selection doesn't do this and I hope it will be obvious to people writing policies that they've requested more reviewers than exist for the teams in use!